azure-toolkit-libs/azure-toolkit-containerapps-lib/src/main/resources/template/source-dockerfile

```diff
@@ -0,0 +1,39 @@
+# Dockerfile.template
+
+# Build-time variable (can be passed using --build-arg)
+ARG JAVA_VERSION=${JAVA_VERSION:-11}
+
+# TODO(ruoyuwang): use microsoft java image.
+# Step 1: Use a Maven image to build the application
+FROM maven:3-eclipse-temurin-${JAVA_VERSION} AS builder
```

Comment:
Why we are not using the Mariner based images?

Reply:
the same as above one.